client/src/main/java/org/threadly/litesockets/client/http/HTTPClient.java

```diff
@@ -93,12 +98,34 @@ public HTTPClient() {
    * @param maxResponseSize the maximum responseSize clients are allowed to send.
    */
   public HTTPClient(int maxConcurrent, int maxResponseSize) {
+    this(maxConcurrent, maxResponseSize, -1);
+  }
+
+  /**
+   * <p>This constructor will let you set the max Concurrent Requests and max Response Size but will still 
+   * create its own {@link SingleThreadScheduler} to use as a threadpool.</p>
+   *  
+   * <p>The maximum queue length parameter can help improved conditions where you want to fail fast 
+   * if the downstream service is fully consumed.  Since all requests will start in the queue we 
+   * recommend to either provide a {@code 0} to leave the queue unbounded, or to set to at least 
+   * the {@code maxConcurrent} value.</p>
+   * 
+   * @param maxConcurrent maximum number of requests to run simultaneously. 
+   * @param maxResponseSize the maximum responseSize clients are allowed to send.
+   * @param maxQueueSize Maximum queue size, {@code <= 0} to leave unbounded.  Recommended to be >= {@code maxConcurrent}
+   */
+  public HTTPClient(int maxConcurrent, int maxResponseSize, int maxQueueSize) {
     this.maxConcurrent = maxConcurrent;
     this.maxResponseSize = maxResponseSize;
     sts = new SingleThreadScheduler();
     this.ssi = sts;
     ntse = new NoThreadSocketExecuter();
     sei = ntse;
+    if (maxQueueSize < 1 || maxQueueSize == Integer.MAX_VALUE) {
```

Comment:
Feel like we should:
```java
if(maxQueueSize > 0 && maxQueueSize < maxConcurrent) {
  maxQueueSize = maxConcurrent
}
```

Not strictly necessary but would be odd to have max concurrent of 10 and queue size of 1, and on the 2nd/3rd request have it be rejected.

Reply:
I was unsure about this.  I considered the same.  I decided to just document how the queue size _generally_ should be >= the max concurrency.

The reason be is one specific use case I can see, which is a case where you have a low throughput client.  A client where requests are rare, but potentially slow.  There you may want to allow more concurrent requests than queued.

A queue length of 1 is still unlikely to make sense, but I thought I would give people the freedom.  I suspect most people wont limit the queue and instead will go with the default we had anyways.